### ISSUE TITLE
x/mobile: gomobile apps using syscall.Lstat() are blocked by seccomp on Android O and P

### ISSUE THREAD

**Issue by Qheb** [golang/go#27797]
### What version of Go are you using (`go version`)?
go version go1.11 linux/amd64

### Does this issue reproduce with the latest release?
Yes, reproduced with go1.11 and go1.10.4

### What operating system and processor architecture are you using (`go env`)?
Android emulator API 26 to 28 (Android O and P) arch x86_64. The issue does not seem to occur on real Android devices (arch ARM).

### What did you do?
Calling anything using `syscall.Lstat()` from the Go code.
For example `ioutil.ReadDir()`, `os.Rename`, `os.RemoveAll`, ...

### What did you expect to see?
No crash

### What did you see instead?
A crash due to Android seccomp filter:
```
F  *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
F  Build fingerprint: 'Android/sdk_gphone_x86_64/generic_x86_64:8.0.0/OSR1.180418.004/4931640:userdebug/dev-keys'
F  Revision: '0'
F  ABI: 'x86_64'
F  pid: 16035, tid: 16061, name: Thread-7
F  signal 31 (SIGSYS), code 1 (SYS_SECCOMP), fault addr --------
F  Cause: seccomp prevented call to disallowed x86_64 system call 0
F      rax 0000000000000006  rbx 0000000000000000  rcx ffffffffffffffff  rdx 0000000000000000
F      rsi 000000c000592038  rdi 000000c0013ac150
F      r8  0000000000000000  r9  0000000000000000  r10 0000000000000000  r11 0000000000000202
F      r12 ffffffffffffffff  r13 0000000000000004  r14 0000000000000003  r15 0000000000000049
F      cs  0000000000000033  ss  000000000000002b
F      rip 00007d075a225be0  rbp 000000c000dd4e00  rsp 000000c000dd4da0  eflags 0000000000000202
F  backtrace:
syscall.Syscall
/usr/local/go/src/syscall/asm_linux_amd64.s:27
os.lstatNolog
/usr/local/go/src/os/stat_unix.go:42
os.Lstat
/usr/local/go/src/os/stat.go:22
os.RemoveAll
/usr/local/go/src/os/path.go:75
[...]
```

This issue is quite similar to #20409 with `lstat(2)` instead of `select(2)`:
- Android 8.0 (API level 26) applies a Secure Computing (SECCOMP) filter to all apps. The list of allowed syscalls is restricted to those exposed through bionic. (https://developer.android.com/about/versions/oreo/android-8.0-changes#security-all)
- Bionic `lstat(2)` is actually using `fstatat(2)` (https://android.googlesource.com/platform/bionic.git/+/master/libc/bionic/lstat.cpp)
- Golang uses `lstat(2)`, which is blocked by seccomp.

Therefore, I think a similar fix would be to replace `lstat(2)` with `fstatat(2)` in Golang.
This seems to have been already done in https://golang.org/src/syscall/syscall_linux_arm64.go (line 47), this explains why the issue is not reproduced on real Android devices. But the issue still exists on emulators which uses the x86_64 architecture.

**Comment by steeve** [comment]
I can reproduce the issue too.

**Comment by steeve** [comment]
@Qheb I don't see the change in https://golang.org/src/syscall/syscall_linux_386.go though, and it yet it works.

**Comment by bcmills** [comment]
CC: @eliasnaur @hyangah

**Comment by steeve** [comment]
that was fast

**Comment by Qheb** [comment]
> @Qheb I don't see the change in https://golang.org/src/syscall/syscall_linux_386.go though, and yet it works.

@steeve I didn't notice, but SECCOMP is actually not blocking `lstat` on x86. I think the important change in https://golang.org/src/syscall/syscall_linux_386.go is at line 66:
```
//sys	Lstat(path string, stat *Stat_t) (err error) = SYS_LSTAT64
```
`lstat64` seems to be used instead of `lstat`. And it turns out that `lstat64` is in one of the [SECCOMP whitelists](https://android.googlesource.com/platform/bionic.git/+/master/libc/SECCOMP_WHITELIST_APP.TXT#46) for arm, x86 and mips architectures:
```
int	lstat64:lstat64(const char*, struct stat64*)	arm,x86,mips
```

Anyway, that was indeed fast, thank you @eliasnaur for this fix.
